service/watcher/configmap/cache.go

```diff
@@ -0,0 +1,213 @@
+package configmap
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/giantswarm/apiextensions/v2/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+
+	applabel "github.com/giantswarm/app-operator/v2/pkg/label"
+	"github.com/giantswarm/app-operator/v2/service/controller/key"
+)
+
+func (c *AppValue) buildCache(ctx context.Context) error {
+	for {
+		lo := metav1.ListOptions{
+			LabelSelector: c.selector.String(),
+		}
+
+		res, err := c.k8sClient.G8sClient().ApplicationV1alpha1().Apps("").Watch(ctx, lo)
+		if err != nil {
+			panic(err)
+		}
+
+		for r := range res.ResultChan() {
+			cr, err := key.ToCustomResource(r.Object)
+			if err != nil {
+				panic(err)
+			}
+
+			err = c.addCache(ctx, cr, r.Type)
+			if err != nil {
+				c.logger.Log("level", "info", "message", "failed to reconcile an app CR", "stack", fmt.Sprintf("%#v", err))
```

Comment:
```suggestion
				c.logger.Log("level", "info", "message", fmt.Sprintf("failed to reconcile app CR %#q", cr.Name), "stack", fmt.Sprintf("%#v", err))
```

Can we log which app CR it is?

Reply:
This is better, thanks.